### PR TITLE
[FEATURE] S3 store backend - add possibility to access bucket assuming iam role

### PR DESCRIPTION
**Pull request by ext-gvillafane**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

## Changes proposed in this pull request:
- Accept role_arn attribute to be able to reach S3 buckets through assume role operation

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by talagluck** [comment]
Hi @ext-gvillafane - thank you so much for your contribution! Someone on the team will review and be in touch. In the meantime, it looks like the PR is failing a linting check. Could you please run the linters as we describe in our contribution docs and re-commit? Thanks!

File: great_expectations/data_context/store/tuple_store_backend.py [hunk 1]
```diff
@@ -722,6 +729,29 @@ def _create_client(self):
...
+            RoleArn=self.role_arn, RoleSessionName="AssumedRoleSession"
+        )
+
+        # From the response that contains the assumed role, get the temporary
+        # credentials that can be used to make subsequent API calls
+        self._boto3_options["aws_access_key_id"] = assumed_role_object["Credentials"]["AccessKeyId"]
+        self._boto3_options["aws_secret_access_key"] = assumed_role_object["Credentials"]["SecretAccessKey"]
+        self._boto3_options["aws_session_token"] = assumed_role_object["Credentials"]["SessionToken"]
+
+        logger.debug(f"Role {self.role_arn} assumed, access keys configured.")
```
Comment (kenwade4) [inline great_expectations/data_context/store/tuple_store_backend.py:752]: Did you mean to put `self.role_assumed = True` somewhere in this `_assume_role` method? If not, feel free to delete the variable from the `__init__` method

Comment (ext-gvillafane) [reply great_expectations/data_context/store/tuple_store_backend.py:752]: oh yes in a first version I used that.. was left there unintentionally. will delete, thanks!

Comment (kenwade4) [reply great_expectations/data_context/store/tuple_store_backend.py:752]: Sorry for the delay @ext-gvillafane. Can you pull latest develop into this branch?

**Comment by anthonyburdi** [comment]
Hi @ext-gvillafane! Thank you for this contribution. I was just trying to make the changes that @kenwade4 suggested (lint / merge develop) but unfortunately I'm not able to push to your branch. Since this PR has been outstanding for a while I will close it but if you can either make those edits or open the branch so I can make those changes we can reopen and get this change in. Thank you!

**Pull request closed without merging**